### ISSUE TITLE
Fix S2758: false positive on conditional operator with interpolated string

### ISSUE THREAD

**Issue by BrightLight** [SonarSource/sonar-csharp#828]
### Description

The following code gets flagged by S2758: "This operation returns the same value whether the condition is 'true' or 'false'" (this code is part of a class that generates a code file)

![image](https://user-images.githubusercontent.com/1859278/31534367-ea87b02a-aff6-11e7-936a-42524aecff4c.png)

This seems like a false positive. I think it returns either
"return this.somefieldname;"
or
"return this.somefieldname.Value;"
These are two different strings.

### Repro steps

Source code to copy into project for scanning:
```csharp
var fieldName = this.GetGeneratedFieldName(member, generationStrategy);
return !this.HasDependentForeignKeyProperty(member)
    ? $"return this.{fieldName};"
    : $"return this.{fieldName}.Value;";
```

### Expected behavior
No issue is created as this clearly returns two different strings.

### Actual behavior
S2758: "This operation returns the same value whether the condition is 'true' or 'false'" is reported

### Related information
* SonarC# Version 6.4.1 (build 3596)
* MSBuild 14.0.23107.0
* Scanner for MSBuild 3.0.2.656 (via Jenkins)

**Comment by Evangelink** [comment]
Hi @BrightLight,
Thanks for the feedback. We are able to reproduce the issue with latest version using your code sample.

**Comment by Evangelink** [comment]
Hi @BrightLight,

We currently rely on Roslyn to detect whether the 2 parts are equals or not. The problem is that we are still using Roslyn 1.0 which doesn't handle interpolated strings hence this False Positive.
We are planning to move to newer versions of Roslyn soon (to support .Net Core) and so this ticket will be automatically fixed.

So I am closing the ticket as we won't do anything on older versions.
